components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketHandler.java

```diff
@@ -0,0 +1,437 @@
+import io.netty.handler.codec.http.websocketx.WebSocketFrame;
+import org.apache.axiom.util.UIDGenerator;
+import org.apache.http.HttpHeaders;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.apimgt.api.APIConsumer;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.AccessTokenInfo;
+import org.wso2.carbon.apimgt.api.model.Subscriber;
+import org.wso2.carbon.apimgt.gateway.handlers.security.APISecurityConstants;
+import org.wso2.carbon.apimgt.gateway.handlers.security.APISecurityException;
+import org.wso2.carbon.apimgt.gateway.handlers.throttling.APIThrottleConstants;
+import org.wso2.carbon.apimgt.gateway.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.gateway.throttling.publisher.ThrottleDataPublisher;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.APIManagerFactory;
+import org.wso2.carbon.apimgt.impl.dao.constants.SQLConstants;
+import org.wso2.carbon.apimgt.impl.dto.APIKeyValidationInfoDTO;
+import org.wso2.carbon.apimgt.impl.factory.KeyManagerHolder;
+import org.wso2.carbon.apimgt.impl.utils.APIMgtDBUtil;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.net.InetSocketAddress;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+//import org.wso2.carbon.inbound.endpoint.osgi.service.ServiceReferenceHolder;
+
+public class WebsocketHandler extends ChannelInboundHandlerAdapter {
+	private static String tenantDomain;
+	private static int port;
+	private static volatile ThrottleDataPublisher throttleDataPublisher = null;
+	private static Logger log = LoggerFactory.getLogger(WebsocketHandler.class);
+	private static ServiceReferenceHolder serviceReferenceHolder;
+	API api;
+	private String uri;
+	private APIConsumer apiConsumer;
+	private Subscriber apiSubscriber;
+	private APIKeyValidationInfoDTO infoDTO = new APIKeyValidationInfoDTO();
+
+	public WebsocketHandler() {
+		System.out.println("executed hand");
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+
+		if (msg instanceof FullHttpRequest) {
+			FullHttpRequest req = (FullHttpRequest) msg;
+			uri = req.getUri();
+			port = ((InetSocketAddress) ctx.channel().localAddress()).getPort();
+			tenantDomain = MultitenantUtils.getTenantDomainFromUrl(req.getUri());
+			if (tenantDomain.equals(req.getUri())) {
+				tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+			}
+
+			if (oauthHandler(req)) {
+				System.out.println("http new");
```

Comment:
remove all system print outs and use logs. Also add debugs logs so it will more useful when troubleshooting issues